Content.Shared/Body/Systems/SharedBodySystem.Body.cs

```diff
@@ -263,29 +271,47 @@ public IEnumerable<BodyPartSlot> GetBodyAllSlots(EntityUid bodyId, BodyComponent
         }
     }
 
-    public virtual HashSet<EntityUid> GibBody(EntityUid bodyId, bool gibOrgans = false,
-        BodyComponent? body = null, bool deleteItems = false, bool deleteBrain = false)
+    public virtual HashSet<EntityUid> GibBody(
+        EntityUid bodyId,
+        bool gibOrgans = false,
+        BodyComponent? body = null ,
+        bool deleteItems = false,
+        bool launchGibs = true,
+        Vector2? splatDirection = null,
+        float splatModifier = 1,
+        Angle splatCone = default,
+        SoundSpecifier? gibSound = null,
+        AudioParams? gibAudioParams = null
+        )
     {
         var gibs = new HashSet<EntityUid>();
 
         if (!Resolve(bodyId, ref body, false))
             return gibs;
 
+        var root = GetRootPartOrNull(bodyId, body);
+        if (root != null && TryComp(root.Value.Entity, out GibbableComponent? gibbable))
+        {
+            gibSound ??= gibbable.GibSound;
+            gibAudioParams ??= gibbable.GibAudioParams;
+        }
         var parts = GetBodyChildren(bodyId, body).ToArray();
         gibs.EnsureCapacity(parts.Length);
-
         foreach (var part in parts)
         {
-            SharedTransform.AttachToGridOrMap(part.Id);
-            gibs.Add(part.Id);
+
+            _gibbingSystem.TryGibEntityWithRef(bodyId, part.Id, GibType.Gib, GibContentsOption.Skip, ref gibs,
+                playAudio: false, launchGibs:true, launchDirection:splatDirection, launchImpulse: GibletLaunchImpulse * splatModifier,
+                launchImpulseVariance:GibletLaunchImpulseVariance, launchCone: splatCone);
 
             if (!gibOrgans)
                 continue;
 
             foreach (var organ in GetPartOrgans(part.Id, part.Component))
             {
-                SharedTransform.AttachToGridOrMap(organ.Id);
-                gibs.Add(organ.Id);
+                _gibbingSystem.TryGibEntityWithRef(bodyId, organ.Id, GibType.Drop, GibContentsOption.Skip,
+                    ref gibs, playAudio: false, launchImpulse: GibletLaunchImpulse* splatModifier,
+                    launchImpulseVariance:GibletLaunchImpulseVariance, launchCone: splatCone);
```

Comment:
```suggestion
                _gibbingSystem.TryGibEntityWithRef(bodyId, organ.Id, GibType.Drop, GibContentsOption.Skip,
                    ref gibs, playAudio: false, launchImpulse: GibletLaunchImpulse * splatModifier,
                    launchImpulseVariance:GibletLaunchImpulseVariance, launchCone: splatCone);
```

Reply:
I always fuck this up lol, I really need to add this to rider's auto format.